bot/code_review_bot/tasks/default.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+import structlog
+from libmozdata.phabricator import LintResult
+
+from code_review_bot import Issue
+from code_review_bot import taskcluster
+from code_review_bot.tasks.base import AnalysisTask
+
+logger = structlog.get_logger(__name__)
+
+ISSUE_MARKDOWN = """
+## issue {analyzer}
+
+- **Path**: {path}
+- **Level**: {level}
+- **Check**: {check}
+- **Line**: {line}
+- **Publishable**: {publishable}
+
+```
+{message}
+```
+"""
+
+
+class DefaultIssue(Issue):
+    ANALYZER = "default"
+
+    def validates(self):
+        """
+        Default issues are valid as long as they match the format
+        """
+        return True
+
+    def as_text(self):
+        """
+        Build the text content for reporters
+        """
+        return "{}: {} [{}]".format(self.level.capitalize(), self.message, self.check)
+
+    def as_markdown(self):
+        """
+        Build the Markdown content for debug email
+        """
+        return ISSUE_MARKDOWN.format(
+            analyzer=self.analyzer,
+            path=self.path,
+            check=self.check,
+            level=self.level,
+            line=self.line,
+            message=self.message,
+            publishable=self.is_publishable() and "yes" or "no",
+        )
+
+    def as_phabricator_lint(self):
+        """
+        Outputs a Phabricator lint result
+        """
+        return LintResult(
+            name=f"Issue {self.analyzer}",
+            description=self.message,
+            code=self.check,
+            severity=self.level,
+            path=self.path,
+            line=self.line,
+            char=self.column,
+        )
+
+
+class DefaultTask(AnalysisTask):
+    """
+    Support issues using the code review format
+    https://github.com/mozilla/code-review/blob/master/docs/analysis_format.md
+    """
+
+    artifacts = ["public/code-review/issues.json"]
+
+    def parse_issues(self, artifacts, revision):
+        """
+        Parse issues from a log file content
+        """
+        assert isinstance(artifacts, dict)
+        return [
+            DefaultIssue(
+                analyzer=issue.get("analyzer", self.name),
+                revision=revision,
+                path=issue["path"],
+                line=issue["line"],
+                column=issue["column"],
+                nb_lines=issue.get("nb_lines", 1),
+                level=issue["level"],
+                check=issue["check"],
```

Comment:
Maybe we should default the check to the analyzer name like we do for MozLint (I guess many analyzers will be single-rule analyzers).

Reply:
If you do this, don't forget to update the docs too

Reply:
Done